src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java

```diff
@@ -3014,6 +3000,70 @@ public int getTier() {
 		new ItemStack[] {new ItemStack(Material.RED_DYE), SlimefunItems.MAGIC_LUMP_3, new ItemStack(Material.CYAN_DYE), new ItemStack(Material.WHITE_WOOL), SlimefunItems.RUNE_ENDER, new ItemStack(Material.WHITE_WOOL), new ItemStack(Material.YELLOW_DYE), SlimefunItems.ENDER_LUMP_3, new ItemStack(Material.MAGENTA_DYE)})
 		.register(true);
 
+        new SlimefunItem(Categories.LUMPS_AND_MAGIC, SlimefunItems.RUNE_SOULBOUND, "ANCIENT_RUNE_SOULBOUND", RecipeType.ANCIENT_ALTAR,
+        new ItemStack[] {SlimefunItems.MAGIC_LUMP_3, SlimefunItems.ESSENCE_OF_AFTERLIFE, SlimefunItems.MAGIC_LUMP_3, new ItemStack(Material.ENDER_PEARL), SlimefunItems.RUNE_ENDER, new ItemStack(Material.ENDER_PEARL), SlimefunItems.MAGIC_LUMP_3, SlimefunItems.ESSENCE_OF_AFTERLIFE, SlimefunItems.MAGIC_LUMP_3})
+        .register(true, new ItemDropHandler() {
+			@Override
+			public boolean onItemDrop(PlayerDropItemEvent e, Player p, Item i) {
+				ItemStack item = i.getItemStack();
+				boolean[] boo = {false};
+				if (SlimefunManager.isItemSimiliar(item, SlimefunItems.RUNE_SOULBOUND, true)) {
+					Bukkit.getScheduler().scheduleSyncDelayedTask(SlimefunPlugin.instance, () -> {
+
+						Location l = i.getLocation();
+						Collection<Entity> entites = l.getWorld().getNearbyEntities(l, 1.5, 1.5, 1.5, (entity) -> entity.getType() == EntityType.DROPPED_ITEM);
+						if (!entites.isEmpty()) {
+
+							ItemStack ench = null;
+							Item ent = null;
+							for (Entity entity: entites) {

```

Comment:
Instead of iterating over this collection again, you should just use the Predicate<> used above.